src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
+inline bool any_of(Container&& c, Predicate p) {
+    using std::begin; //ADL
```

Comment:
I'm one of those non-programmer types Kevin talks about.  All the C++ I know I learned from working on DDA.

So I'd have guessed that std::begin and ::end were some sort of weird code things, and, as Kevin pointed out, had to go hunting for reference material.

We take all comers here, so it's hardly "pessimistic" to presume we don't know what ADL means in this context.